src/lsdb/dask/merge_catalog_functions.py

```diff
@@ -18,6 +19,37 @@
     from lsdb.catalog.catalog import Catalog
 
 
+def concat_partition_and_margin(
+    partition: pd.DataFrame, margin: pd.DataFrame | None, right_columns: List[str]
+) -> pd.DataFrame:
+    """Concatenates a partition and margin dataframe together
+
+    Args:
+        partition (pd.DataFrame): The partition dataframe
+        margin (pd.DataFrame): The margin dataframe
+
+    Returns:
+        The concatenated dataframe with the partition on top and the margin on the bottom
+    """
+    hive_columns = [
+        PartitionInfo.METADATA_ORDER_COLUMN_NAME,
+        PartitionInfo.METADATA_DIR_COLUMN_NAME,
+        PartitionInfo.METADATA_PIXEL_COLUMN_NAME,
+    ]
+    margin_filtered = None
+    if margin is not None:
+        margin_columns_no_hive = [col for col in margin.columns if col not in hive_columns]
+        rename_columns = {
+            f"margin_{PartitionInfo.METADATA_ORDER_COLUMN_NAME}": PartitionInfo.METADATA_ORDER_COLUMN_NAME,
+            f"margin_{PartitionInfo.METADATA_DIR_COLUMN_NAME}": PartitionInfo.METADATA_DIR_COLUMN_NAME,
+            f"margin_{PartitionInfo.METADATA_PIXEL_COLUMN_NAME}": PartitionInfo.METADATA_PIXEL_COLUMN_NAME,
+        }
+        margin_renamed = margin[margin_columns_no_hive].rename(columns=rename_columns)
+        margin_filtered = margin_renamed[right_columns]
+    joined_df = pd.concat([partition, margin_filtered]) if margin_filtered is not None else partition
```

Comment:
This method would be easier to read with return early, e.g.:

```
def do_stuff():
  if margin is None:
    return partition
  hive_columns = ...
  ...
  return pd.concat([partition, margin_filtered)]
```

Reply:
Ah that's nicer!